### PR TITLE
Fix Tooltip Issue on Switch

### LINKED ISSUE #110222: `Tooltip` does not display for `TooltipTriggerMode.tap` with `Switch` as child
## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Tap on the button to see the desired behavior for displaying a tooltip
3. Now tap on the switch 

**Expected results:** 
The tooltip is displayed

**Actual results:** 
The tooltip is not displayed
Note that if I use `Switch.adaptive` I can get the expected behavior for iOS, so the Cupertino one seems to be working correctly.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
          appBar: AppBar(title: const Text('Bug - Tooltip for MaterialSwitch')),
          body: Padding(
            padding: const EdgeInsets.all(24.0),
            child: ListView(
              children: [
                const Text('Tooltip for Button - works as expected'),
                const Tooltip(
                  triggerMode: TooltipTriggerMode.tap,
                  message: 'You do not have permission to order muffins.',
                  child: ElevatedButton(
                    child: Text('Order'),
                    onPressed: null,
                  ),
                ),
                const SizedBox(
                  height: 24.0,
                ),
                const Text('Tooltip for Switch - does not work as expected'),
                Row(
                  children: const [
                    Text('Notify group when muffins arrive'),
                    Tooltip(
                      triggerMode: TooltipTriggerMode.tap,
                      message: 'You do not have permission to manage alerts',
                      child: Switch(onChanged: null, value: false),
                    ),
                  ],
                ),
              ],
            ),
          )),
    );
  }
}

```

</details>

<details>
  <summary>Logs</summary>

```
[+1529 ms] E/SurfaceSyncer(13374): Failed to find sync for id=0
[  +21 ms] W/Parcel  (13374): Expecting binder but got null!
```

```
Analyzing bug_tooltip_switch...
No issues found! (ran in 2.2s)
```

```
fvm flutter doctor -v                       ✔  4s
[✓] Flutter (Channel stable, 2.10.4, on macOS 12.5 21G72 darwin-x64, locale en-US)
    • Flutter version 2.10.4 at /Users/lisamartin/Projects/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c860cba910 (5 months ago), 2022-03-25 00:23:12 -0500
    • Engine revision 57d3bac3dd
    • Dart version 2.16.2
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/lisamartin/Library/Android/sdk
    • Platform android-33, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.70.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.46.0

[✓] Connected device (3 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554                        • android-x64    • Android 13 (API 33) (emulator)
    • iPhone 11 Pro Max (mobile)   • EAC08C7E-6BEE-4E68-A70C-3DA080D0430D • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • Chrome (web)                 • chrome                               • web-javascript • Google Chrome 104.0.5112.101

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

https://user-images.githubusercontent.com/6465955/186537516-f1b0117e-f623-46de-b5c9-15fbfd29e7b3.mov

### PR DESCRIPTION
**Pull request by QuncCccccc**
This PR fixes the iOS/Android use case when the Switch is **disabled** and the tooltip's `triggerMode` is set to `TooltipTriggerMode.tap`. The change also affects Checkbox and Radio components which are also created by the `buildToggleable` method.

For these **enabled** components, the default setting(TooltipTriggerMode.longPress) still shows a tooltip message, but if the `triggerMode` is set to `TooltipTriggerMode.tap`, the Tooltip's `GestureDetecter.onTap` will compete with other components' `GestureDetecter.onTap`, and as a result, the tooltip will not show.

https://user-images.githubusercontent.com/36861262/188039144-9fb4685e-c628-4eb6-b885-a4e92e6462a7.mov

Fixes: #110222

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/test/material/checkbox_test.dart [hunk 1]
```diff
@@ -1446,6 +1447,57 @@ void main() {
...
   });
+
+  testWidgets('disabled checkbox shows tooltip', (WidgetTester tester) async {
+    const String longPressTooltip = 'long press tooltip';
+    const String tapTooltip = 'tap tooltip';
+    await tester.pumpWidget(
+        const MaterialApp(
+          home: Material(
+            child: Tooltip(
+              message: 'long press tooltip',
```
Comment (bleroux) [inline packages/flutter/test/material/checkbox_test.dart:1458]:
```suggestion
              message: longPressTooltip,
```

File: packages/flutter/test/material/checkbox_test.dart [hunk 2]
```diff
@@ -1446,6 +1447,57 @@ void main() {
...
+    await tester.pumpWidget(
+        const MaterialApp(
+          home: Material(
+            child: Tooltip(
+              triggerMode: TooltipTriggerMode.tap,
+              message: tapTooltip,
+              child: Checkbox(value: true, onChanged: null),
+            ),
+          ),
+        )
```
Comment (bleroux) [inline packages/flutter/test/material/checkbox_test.dart:1491]: nit: remove extra indentation
(same for the previous call to `pumpWidget` in this file and in the other test files)

File: packages/flutter/test/material/checkbox_test.dart [hunk 3]
```diff
@@ -1446,6 +1447,57 @@ void main() {
...
+
+    final TestGesture gestureLongPress = await tester.startGesture(tester.getCenter(tooltip0));
+    await tester.pump();
+    await tester.pump(kLongPressTimeout);
+    await gestureLongPress.up();
+    await tester.pump();
+
+    expect(find.text(longPressTooltip), findsOneWidget);
+
+    // Tooltip shows up after tapping when set triggerMode to TooltipTriggerMode.longPress.
```
Comment (bleroux) [inline packages/flutter/test/material/checkbox_test.dart:1481]: TooltipTriggerMode.longPress.-> TooltipTriggerMode.tap

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**